### DISCUSSION TITLE
Make the road to contributing better and less painfull ie Building of source locally

### DISCUSSION THREAD

**Discussion by davidbuckleyni** [dotnet/maui#9490] [General]
Even today I tried to compile the Maui source as was going to do a Pull Request for an issue that I found however the build failed.

There need to be more docs on how one can contribute and build the source locally
At present when I attempt to compile in latest preview nothing works over 27000 errors.

Please provide a clear an updated step-by-step guide on how to build the source using public visual studio channels not internal ones.

This does not seem better than a year on no clear instructions on how to set up your machine for building the source. James did a video a few months ago but even himself ran into issues. It should not be this hard to contribute and build a open source project.

![image](https://user-images.githubusercontent.com/5619229/185147722-c0d061ad-f7d4-451f-8b48-6e949768aa62.png)

**Comment by PureWeen** [comment]
@davidbuckleyni did you follow this guide?

https://github.com/dotnet/maui/blob/main/.github/DEVELOPMENT.md

I'm using VS Stable currently everywhere to develop

**Comment by davidbuckleyni** [comment]
Yeah I do use Cake, but one of the other teams members when Xamrain Forms was coming out showed a way to debug your build and I am wondering if something is possible with Maui in the same way it dont feel like a happy path having to do all that? @PureWeen

**Reply by PureWeen** [reply]
Yea you can use the cake script to do that as well
https://github.com/dotnet/maui/blob/main/.github/DEVELOPMENT.md#testing-branch-against-your-project
